src/components/legend/helpers.js

```diff
@@ -0,0 +1,25 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plots = require('../../plots/plots');
+
+
+exports.legendGetsTrace = function legendGetsTrace(trace) {
+    return trace.visible && Plots.traceIs(trace, 'showLegend');
+};
+
+exports.isGrouped = function isGrouped(legendLayout) {
+    return (legendLayout.traceorder || '').indexOf('grouped') !== -1;
+};
+
+exports.isReversed = function isReversed(legendLayout) {
```

Comment:
I personally prefer `var helpers`. 

I think capital letters should only be used to denote objects exported in `index.js` files.